libnavigation-core/src/main/java/com/mapbox/navigation/core/routeoptions/RouteOptionsUpdater.kt

```diff
@@ -100,32 +100,15 @@ class RouteOptionsUpdater {
                                 it.addAll(approachesList.subList(index, coordinatesList.size))
                             }
                         }
-                    ).apply {
-                        if (routeOptions.profile() == DirectionsCriteria.PROFILE_DRIVING_TRAFFIC) {
-                            snappingIncludeClosuresList(
-                                let snappingClosures@{
-                                    val snappingClosures =
-                                        routeOptions.snappingIncludeClosuresList()
-                                    mutableListOf<Boolean?>().apply {
-                                        // append true for the origin of the re-route request
-                                        add(true)
-                                        if (snappingClosures.isNullOrEmpty()) {
-                                            // create `null` value for each upcoming waypoint
-                                            addAll(arrayOfNulls<Boolean>(remainingWaypoints))
-                                        } else {
-                                            // get existing values for each upcoming waypoint
-                                            addAll(
-                                                snappingClosures.subList(
-                                                    index + 1,
-                                                    coordinatesList.size
-                                                )
-                                            )
-                                        }
-                                    }
-                                }
-                            )
-                        }
-                    }
+                    )
+                    .snappingIncludeClosuresList(
+                        routeOptions.snappingIncludeClosuresList()
+                            .withFirstTrue(remainingWaypoints, index, coordinatesList.size)
+                    )
```

Comment:
We cannot rely on nullability when adding `snappingIncludeClosuresList` and `snappingIncludeStaticClosuresList`. We do want to add these parameters to a reroute request even if they were not part of the original request. That's why we need to check for the profile.

Reply:
I returned the profile check so it should work as expected now.